src/operator/map.rs

```diff
@@ -0,0 +1,221 @@
+    }
+
+    /// Apply [`MapKeys`] operator to `self`.
+    ///
+    /// The `map` closure operates on owned keys.
+    pub fn map_keys_owned<K1, K2, V, CO, F>(&self, map: F) -> Stream<Circuit<P>, CO>
+    where
+        K1: Clone + 'static,
+        K2: Clone + 'static,
+        V: Clone + 'static,
+        CI: IntoIterator<Item = (K1, V)> + 'static,
+        for<'a> &'a CI: IntoIterator,
+        for<'a> <&'a CI as IntoIterator>::Item: RefPair<'a, K1, V>,
+        CO: FromIterator<(K2, V)> + Clone + 'static,
+        F: Fn(K1) -> K2 + Clone + 'static,
+    {
+        let func_clone = map.clone();
+        self.circuit().add_unary_operator(
+            MapKeys::new(move |x: &K1| (map)(x.clone()), func_clone),
+            self,
+        )
+    }
+}
+
+/// Operator that applies a user-defined function to each value in a collection
+/// of key/value pairs.
+///
+/// # Type arguments
+///
+/// * `K1` - input key type.
+/// * `K2` - output key type.
+/// * `V` - value type.
+/// * `CI` - input collection type.
+/// * `CO` - output collection type.
+/// * `FB` - key mapping function type that takes a borrowed key.
+/// * `FO` - key mapping function type that takes an owned key.
+pub struct MapKeys<K1, K2, V, CI, CO, FB, FO>
+where
+    FB: 'static,
+    FO: 'static,
+{
+    map_borrowed: FB,
+    map_owned: FO,
+    _type: PhantomData<(K1, K2, V, CI, CO)>,
+}
+
+impl<K1, K2, V, CI, CO, FB, FO> MapKeys<K1, K2, V, CI, CO, FB, FO>
+where
+    FB: 'static,
+    FO: 'static,
+{
+    pub fn new(map_borrowed: FB, map_owned: FO) -> Self {
+        Self {
+            map_borrowed,
+            map_owned,
+            _type: PhantomData,
+        }
+    }
+}
+
+impl<K1, K2, V, CI, CO, FB, FO> Operator for MapKeys<K1, K2, V, CI, CO, FB, FO>
+where
+    K1: 'static,
+    K2: 'static,
+    V: 'static,
+    CI: 'static,
+    CO: 'static,
+    FB: 'static,
+    FO: 'static,
+{
+    fn name(&self) -> Cow<'static, str> {
+        Cow::from("MapKeys")
+    }
+    fn clock_start(&mut self, _scope: Scope) {}
+    fn clock_end(&mut self, _scope: Scope) {}
+}
+
+impl<K1, K2, V, CI, CO, FB, FO> UnaryOperator<CI, CO> for MapKeys<K1, K2, V, CI, CO, FB, FO>
+where
+    K1: Clone + 'static,
+    K2: Clone + 'static,
+    V: Clone + 'static,
+    CI: IntoIterator<Item = (K1, V)> + 'static,
+    for<'a> &'a CI: IntoIterator,
+    for<'a> <&'a CI as IntoIterator>::Item: RefPair<'a, K1, V>,
+    CO: FromIterator<(K2, V)> + 'static,
+    FB: Fn(&K1) -> K2 + 'static,
+    FO: Fn(K1) -> K2 + 'static,
+{
+    fn eval(&mut self, i: &CI) -> CO {
+        i.into_iter()
+            .map(|pair| {
+                let (k, v) = pair.into_refs();
+                ((self.map_borrowed)(k), v.clone())
+            })
+            .collect()
+    }
+
+    fn eval_owned(&mut self, i: CI) -> CO {
+        i.into_iter()
+            .map(|(k, v)| ((self.map_owned)(k), v))
+            .collect()
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::vec;
+
+    use crate::{
```

Comment:
```suggestion
    use std::vec;
    use crate::{
```
I'd also just straight up import `vec::IntoIter` here since it's a test and doesn't clash with anything